pulp_ansible/app/tasks/collections.py

```diff
@@ -105,46 +107,24 @@ def import_collection(
             match the metadata in the tarball.
 
     """
-    artifact = Artifact.objects.get(pk=artifact_pk)
-    log.info("Processing collection from {path}".format(path=artifact.file.path))
-    importer_result = process_collection(str(artifact.file.path))
-    try:
-        collection_info = importer_result["metadata"]
-
-        if expected_namespace and expected_namespace != collection_info["namespace"]:
-            msg = _(
-                "Expected namespace '{expected_namespace}', but instead Collection has namespace "
-                "'{collection_namespace}'."
-            )
-            raise ValidationError(
-                msg.format(
-                    expected_namespace=expected_namespace,
-                    collection_namespace=collection_info["namespace"],
-                )
-            )
 
-        if expected_name and expected_name != collection_info["name"]:
-            msg = _(
-                "Expected name '{expected_name}', but instead Collection has name "
-                "'{collection_name}'."
-            )
-            raise ValidationError(
-                msg.format(expected_name=expected_name, collection_name=collection_info["name"])
-            )
-
-        if expected_version and expected_version != collection_info["version"]:
-            msg = _(
-                "Expected version '{expected_version}', but instead Collection has version "
-                "'{collection_version}'."
-            )
-            raise ValidationError(
-                msg.format(
-                    expected_version=expected_version, collection_version=collection_info["version"]
-                )
-            )
-    except Exception:
-        artifact.delete()
-        raise
+    artifact = Artifact.objects.get(pk=artifact_pk)
+    filename = None
+    if expected_namespace and expected_name and expected_version:
```

Comment:
This adjusts it to require all 3 of them which I can see a use case when a person only asserts on the namespace and not version for example. Can we switch this back?

Reply:
@cutwater @bmbouter if we agree to do that, the importer can validate on only what is passed to it

Reply:
@bmbouter I don't have objections, however it'll require update to importer to expect empty fields in `CollectionFilename` tuple.

Reply:
I agree. I'm going to see if I can work with @awcrosby to coordinate via slack.